pkg/scheduler/framework/types.go

```diff
@@ -1052,19 +1052,73 @@ func (n *NodeInfo) update(pod *v1.Pod, sign int64) {
 	n.Generation = nextGeneration()
 }
 
+// getNonMissingContainerRequests returns the default non-zero CPU and memory
+// requests for a container that the scheduler uses when container-level and
+// pod-level requests are not set for a resource. It returns a ResourceList that
+// includes these default non-zero requests, which are essential for the
+// scheduler to function correctly.
+// The method's behavior depends on whether pod-level resources are set or not:
+// 1. When the pod level resources are not set, the method returns a ResourceList
+// with the following defaults:
+//   - CPU: schedutil.DefaultMilliCPURequest
+//   - Memory: schedutil.DefaultMemoryRequest
+//
+// These defaults ensure that each container has a minimum resource request,
+// allowing the scheduler to aggregate these requests and find a suitable node
+// for the pod.
+//
+// 2. When the pod level resources are set, if a CPU or memory request is
+// missing at the container-level *and* at the pod-level, the corresponding
+// default value (schedutil.DefaultMilliCPURequest or schedutil.DefaultMemoryRequest)
+// is included in the returned ResourceList.
+// Note that these default values are not set in the Pod object itself, they are only used
+// by the scheduler during node selection.
+func getNonMissingContainerRequests(requests v1.ResourceList, podLevelResourcesSet bool) v1.ResourceList {
```

Comment:
@alculquicondor  We used to have the discussion about the default values for CPU and memory is a problem if we have multiple containers for one pod or multiple Pods in a node (I can't find the issue), maybe it's a good chance to have the default values for a Pod rather than each containers?

Reply:
We discussed about the same in sig node; As much as we want to have the pod-level default values as a part of this PR, we are very late given the code freeze is in 2 days! I would like to keep the behavior as-is for now, and address this in next release.

Reply:
I agree that it's too late to change at this point.

Reply:
@alculquicondor please add a link to the discussion. I will add it to my notes, so that I can get to it in the next release.

Reply:
No worries, added this as a TODO in my notes. Resolving the comment for now.

Reply:
Yeah, just a reminder, we can follow this later.